### PR TITLE
Run RivetInterface with different genEventInfoProduct weights

### PR DESCRIPTION
**Pull request by mseidel42**
Needed for generators where weights are not stored in LHE files, e.g. Sherpa.

I suggest to just reuse the existing option `LHEweightNumber` to select which weight is to be used.

This will get another rewrite when Rivet starts to support multiple weights at once for making error bands.

### CODE REVIEW

File: GeneratorInterface/RivetInterface/plugins/RivetAnalyzer.cc [hunk 1]
```diff
@@ -32,10 +32,7 @@ _xsection(-1.)
     if (!pset.exists("GenEventInfoCollection")){
       throw cms::Exception("RivetAnalyzer") << "when using an external event weight you have to specify the GenEventInfoProduct collection from which the weight has to be taken " ; 
     }
-    _LHECollection          = consumes<LHEEventProduct>(pset.getParameter<edm::InputTag>("LHECollection"));
-    _useLHEweights          = pset.getParameter<bool>("useLHEweights");
-    _LHEweightNumber        = pset.getParameter<int>("LHEweightNumber");    
-    
+
```
Comment (intrepid42) [inline GeneratorInterface/RivetInterface/plugins/RivetAnalyzer.cc]: (these lines just clean duplicated garbage)

File: GeneratorInterface/RivetInterface/plugins/RivetAnalyzer.cc [hunk 2]
```diff
@@ -119,7 +116,7 @@ void RivetAnalyzer::analyze(const edm::Event& iEvent,const edm::EventSetup& iSet
       if(!_useLHEweights){
 	edm::Handle<GenEventInfoProduct> genEventInfoProduct;
 	iEvent.getByToken(_genEventInfoCollection, genEventInfoProduct);
-	tmpGenEvtPtr->weights()[0] = genEventInfoProduct->weight();
+	tmpGenEvtPtr->weights()[0] = genEventInfoProduct->weights().at(_LHEweightNumber);
```
Comment (perrozzi) [inline GeneratorInterface/RivetInterface/plugins/RivetAnalyzer.cc]: isn't "LHE" a bit misleading here?
ps what is the use case? weights from pythia?

**Comment by intrepid42** [comment]
We could rename the parameter to "weightNumber" but then peoples' 
configurations are broken...

I tested this for weights in Sherpa and Herwig7 (PS weights) so far.

**Comment by perrozzi** [comment]
I think some non-compatibility is part of normal code development...
However I wonder if we want to use exactly the same argument for 2 distinct features.
In principle one would like to be able to apply independently LHE and PS weights and combine them if necessary.
Can you consider adding independent parameters instead of recycling exiting ones?

**Comment by perrozzi** [comment]
ping @intrepid42

**Comment by intrepid42** [comment]
Added new parameters for the GEN weights, so old configs using LHE weights should still work. When both GEN and LHE weights are used, they are multiplied.

**Comment by perrozzi** [comment]
please test

**Comment by perrozzi** [comment]
+1

**Comment by davidlange6** [comment]
merge

**Pull request merged**